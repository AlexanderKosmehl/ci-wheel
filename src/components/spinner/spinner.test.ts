import {
  describe, expect, it,
} from 'vitest';
import generateSpinner from './spinner';

describe('generateSpinner', () => {
  const labels = ['Test', 'Test2'];

  const spinner = generateSpinner({
    labels,
  });

  it('generates component correctly', () => {
    expect(spinner).toMatchSnapshot();
  });
});
