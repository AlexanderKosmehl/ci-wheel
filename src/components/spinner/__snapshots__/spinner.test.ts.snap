// Vitest Snapshot v1

exports[`generateSpinner > generates component correctly 1`] = `
<div
  class="_spinner_5021d3"
  style="transition: 2s;"
>
  <div
    class="_segment_1f8024"
    style="transform: rotate(0deg); clip-path: polygon(
      calc(50% + 5px) 0%,
      100% 0%,
      100% 100%,
      calc(50% + 5px) 100%
    ); background-color: rgb(0, 190, 255);"
  >
    <span
      class="_label_1f8024"
      style="width: calc(50% - 1.5rem - 48px);"
      title="Test"
    >
      Test
    </span>
  </div>
  <div
    class="_segment_1f8024"
    style="transform: rotate(180deg); clip-path: polygon(
      calc(50% + 5px) 0%,
      100% 0%,
      100% 100%,
      calc(50% + 5px) 100%
    ); background-color: rgb(255, 19, 20);"
  >
    <span
      class="_label_1f8024"
      style="width: calc(50% - 1.5rem - 48px);"
      title="Test2"
    >
      Test2
    </span>
  </div>
</div>
`;
